### PR TITLE
staticpod: wait for PodReady condition

### PR DESCRIPTION
**Pull request by sttts**
library-go part of https://github.com/openshift/cluster-kube-apiserver-operator/pull/112.

### CODE REVIEW

File: pkg/operator/staticpod/controller/installer/installer_controller.go [hunk 1]
```diff
@@ -78,23 +94,88 @@ func NewInstallerController(
 	return c
 }
 
+func (c *InstallerController) getStaticPodState(nodeName string) (state staticPodState, deploymentID string, errors []string, err error) {
+	pod, err := c.kubeClient.CoreV1().Pods(c.targetNamespace).Get(c.staticPodName+"-"+nodeName, metav1.GetOptions{})
```
Comment (mfojtik) [inline pkg/operator/staticpod/controller/installer/installer_controller.go]: please convert `c.staticPodName+"-"+nodeName` to `getStaticPodName()`

File: pkg/operator/staticpod/controller/installer/installer_controller.go [hunk 2]
```diff
@@ -78,23 +94,88 @@ func NewInstallerController(
...
+		if apierrors.IsNotFound(err) {
+			return staticPodStatePending, "", nil, nil
+		}
+		return staticPodStatePending, "", nil, err
+	}
+	switch pod.Status.Phase {
+	case corev1.PodRunning, corev1.PodSucceeded:
+		for _, c := range pod.Status.Conditions {
+			if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+				return staticPodStateReady, pod.Labels[deploymentIDLabel], nil, nil
```
Comment (mfojtik) [inline pkg/operator/staticpod/controller/installer/installer_controller.go]: check this label exists?

File: pkg/operator/staticpod/controller/installer/installer_controller.go [hunk 3]
```diff
@@ -78,23 +94,88 @@ func NewInstallerController(
...
+				startNode = i
+				break
+			}
+		}
+	}
+
+	return startNode, nil
+}
+
 // createInstallerController_v311_00_to_latest takes care of creating content for the static pods to deploy.
```
Comment (mfojtik) [inline pkg/operator/staticpod/controller/installer/installer_controller.go]: nit: sync witch function name

File: pkg/operator/staticpod/controller/installer/installer_controller.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/staticpod/controller/installer/installer_controller.go]: done

File: pkg/operator/staticpod/controller/installer/installer_controller.go (hunk 2 above)
Comment (sttts) [reply pkg/operator/staticpod/controller/installer/installer_controller.go]: why? This returns `""` which is correct.

**Pull request merged**